GoRogue.Debugger/Routines/PolygonRoutine.cs

```diff
@@ -0,0 +1,150 @@
+                _map[pos] = TileState.Wall;
+
+            _polygons.Add(Polygon(_cornerAmount));
+
+            // Update map values based on regions
+            ApplyPolygonsToMap();
+        }
+
+        private PolygonArea Polygon(int cornerCount)
+        {
+            var midPoint = new Point(_map.Width / 2, _map.Height / 2);
+            var dim = _map.Width < _map.Height ? _map.Width : _map.Height;
+            var corners = new List<Point>();
+            bool star = false;
+            bool inner = false;
+            int radius;
+            //draw simple shapes up to hexagons
+            if(cornerCount >= 7)
+            {
+                cornerCount *= 2;
+                star = true;
+            }
+
+            for (int i = 0; i < 360; i += 360 / cornerCount)
+            {
+                //each full rotation, increase the distance from the center we're placing our polygons
+                if (star)
+                {
+                    if (inner)
+                        radius = dim / 4;
+                    else
+                        radius = dim * 3 / 4;
+
+                    inner = !inner;
+                }
+                else
+                    radius = dim * 3 / 4;
+
+                var corner = new PolarCoordinate(radius, SadRogue.Primitives.MathHelpers.ToRadian(i)).ToCartesian();
+                corner += midPoint;
+                corners.Add(corner);
+            }
+
+            return new PolygonArea(corners);
+        }
+
+        // Update map to effectively revert all region positions back to walls
+        private void RemovePolygonsFromMap()
+        {
+            foreach (var polygon in _polygons)
+                foreach (var point in polygon.Where(point => _map.Contains(point)))
+                    _map[point] = TileState.Wall;
+        }
+
+        // Apply proper values to map, based on regions
+        private void ApplyPolygonsToMap()
+        {
+            foreach (var polygon in _polygons)
+            {
+                foreach (var point in polygon.InnerPoints.Where(point => _map.Contains(point)))
+                    _map[point] = TileState.InnerRegionPoint;
+
+                foreach (var point in polygon.OuterPoints.Where(point => _map.Contains(point)))
+                    _map[point] = TileState.OuterRegionPoint;
+
+                foreach (var point in polygon.Corners.Where(point => _map.Contains(point)))
+                    _map[point] = TileState.Door;
+            }
+        }
+
+        // Translate point to character based on whether or not the point is in the InnerPoints of a region,
+        // in the OuterPoints of a region, or not in any region
+        private char RegionsView(Point pos)
+            => _map[pos] switch
+            {
+                TileState.InnerRegionPoint => '.',
+                TileState.OuterRegionPoint => '+',
+                TileState.Door => '#', //actually a corner, but that enum has enough junk in it
```

Comment:
I took a more wholistic view of how I'm using the enum, and decide that TileState.Wall, TileState.Floor, and TileState.Door didn't belong in any of my calculation, nor were any of my enum values required by your routines.

So, I removed my enum values from that and made more specific enums for each of my routines.

I didn't touch any routines you wrote, because they all seemed like they were using wall, floor, and door as you intended.